app/scripts/views/sign_in_token_code.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Allow the user to unblock their signin by entering
+ * in a verification code that is sent in an email.
+ */
+define(function (require, exports, module) {
+  'use strict';
+
+  const Cocktail = require('cocktail');
+  const Constants = require('../lib/constants');
+  const FormView = require('./form');
+  const ResendMixin = require('./mixins/resend-mixin')();
+  const SignInMixin = require('./mixins/signin-mixin');
+  const Template = require('stache!templates/sign_in_token_code');
+  const VerificationReasonMixin = require('./mixins/verification-reason-mixin');
+
+  const View = FormView.extend({
+    className: 'sign-in-token-code',
+    template: Template,
+
+    getAccount () {
+      return this.model.get('account');
+    },
+
+    beforeRender () {
+      // user cannot confirm if they have not initiated a sign in.
+      if (! this.model.get('account')) {
+        this.navigate(this._getAuthPage());
+      }
+    },
+
+    setInitialContext (context) {
+      const email = this.getAccount().get('email');
+      const supportLink = this._getSupportLink();
+
+      context.set({
+        email,
+        escapedSupportLink: encodeURI(supportLink),
+        hasSupportLink: !! supportLink
+      });
+    },
+
+    submit () {
+      const account = this.getAccount();
+      const code = this.getElementValue('#token_code');
+      return account.verifyTokenCode(code)
+        .then(() => {
+          return this.invokeBrokerMethod('afterCompleteSignInTokenCode', account);
+        }, (err) => {
+          this.displayError(err);
+        });
+    },
+
+    /**
+     * Get the URL of the page for users that
+     * must enter their password.
+     *
+     * @returns {String}
+     */
+    _getAuthPage () {
+      const authPage =
+        this.model.get('lastPage') === 'force_auth' ? 'force_auth' : 'signin';
+
+      return this.broker.transformLink(authPage);
+    },
+
+    /**
+     * Get the SUMO link for `Why is this happening to me?`. Could be
+     * `undefined` if no link is available.
+     *
+     * @returns {String}
+     */
+    _getSupportLink () {
```

Comment:
nit: why do we need this function? I know other views have this, but can it just be `const supportLink = Constants.BLOCKED_SIGNIN_SUPPORT_URL` above?